### PR TITLE
Add users metadata to user info token

### PR DESCRIPTION
**Pull request by mkunz-intension**

### CODE REVIEW

File: user-info-api/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# API definition for users metadata
```
Comment (ingokuba) [inline user-info-api/README.md:1]: Can you reference the READMEs of the `user-info-*` projects in the parent README?
Best case summarize the three in one Markdown? :thinking:

File: user-info-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>fwu-kc-extensions</artifactId>
+    <groupId>de.intension</groupId>
+    <version>${projectVersion}</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>user-info-api</artifactId>
```
Comment (ingokuba) [inline user-info-api/pom.xml:12]: Maybe rename all folders/names to `person-info`? Better now than later

File: user-info-api/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+|person.kontext.rolle|String|Rolle der Person|
+|person.kontext.status|String|Status|
+### Multiple contexts
+
+Context array
+```
+<zero> ::= 0
+<posDigit> ::= 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9
+<digit> ::= <zero> | <posDigit>
+<number> ::= <zero> <posDigit> | <posDigit> <digit>
```
Comment (ingokuba) [inline user-info-api/README.md:43]: What does this mean? :sweat_smile:

**Review by ingokuba — CHANGES_REQUESTED** [review]
The most important thing is the naming - future Devs will be confused as hell (Easy fix)

File: user-info-api/src/main/java/de/intension/api/UserInfoAttribute.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    PERSON_AKRONYM(UserInfoAttributeNames.PERSON_AKRONYM, UserInfoAttributeNames.PERSON_AKRONYM, Boolean.FALSE, null),
+    PERSON_GEBURTSDATUM(UserInfoAttributeNames.PERSON_GEBURTSDATUM, UserInfoAttributeNames.PERSON_GEBURTSDATUM, Boolean.FALSE, null),
+    PERSON_ALTER(UserInfoAttributeNames.PERSON_ALTER, UserInfoAttributeNames.PERSON_ALTER, Boolean.FALSE, null),
+    PERSON_GESCHLECHT(UserInfoAttributeNames.PERSON_GESCHLECHT, UserInfoAttributeNames.PERSON_GESCHLECHT, Boolean.FALSE, null),
+    PERSON_LOKALISIERUNG(UserInfoAttributeNames.PERSON_LOKALISIERUNG, UserInfoAttributeNames.PERSON_LOKALISIERUNG, Boolean.TRUE, "de-DE"),
+    PERSON_VERTRAUENSSTUFE(UserInfoAttributeNames.PERSON_VERTRAUENSSTUFE, UserInfoAttributeNames.PERSON_VERTRAUENSSTUFE, Boolean.TRUE, Vertrauensstufe.VOLL),
+
+    PERSON_KONTEXT_ID(UserInfoAttributeNames.PERSON_KONTEXT_ID, UserInfoAttributeNames.PERSON_KONTEXT_ID, Boolean.TRUE, Vertrauensstufe.VOLL),
+    PERSON_KONTEXT_ORG_VIDIS_ID(UserInfoAttributeNames.PERSON_KONTEXT_ORG_VIDIS_ID, UserInfoAttributeNames.PERSON_KONTEXT_ORG_VIDIS_ID, Boolean.TRUE, null),
```
Comment (ingokuba) [inline user-info-api/src/main/java/de/intension/api/UserInfoAttribute.java:27]: A tad long maybe :shrug: 
image

File: user-info-provider/src/main/java/de/intension/mapper/oidc/UserInfoProviderMapper.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    /**
+     * Add custom configuration entries.
+     */
+    private static void addConfigEntry(UserInfoAttribute attribute)
+    {
+        ProviderConfigProperty property = new ProviderConfigProperty();
+        property.setName(attribute.getAttributeName());
+        property.setLabel(attribute.getLabel());
+        property.setDefaultValue(attribute.isEnabled().toString());
+        property.setType(ProviderConfigProperty.BOOLEAN_TYPE);
```
Comment (ingokuba) [inline user-info-provider/src/main/java/de/intension/mapper/oidc/UserInfoProviderMapper.java:111]: Can you add a help text explaining what they're for? :thinking: 
image

File: user-info-api/pom.xml (hunk 2 above)
Comment (mkunz-intension) [reply user-info-api/pom.xml:12]: Discussed with the customer -> we will keep it as it is

File: user-info-api/README.md (hunk 3 above)
Comment (mkunz-intension) [reply user-info-api/README.md:43]: BNF

File: user-info-api/src/main/java/de/intension/api/UserInfoAttribute.java (hunk 4 above)
Comment (mkunz-intension) [reply user-info-api/src/main/java/de/intension/api/UserInfoAttribute.java:27]: this is an issue with the Keycloak 20 GUI. In 16.1.1 it was fine.
We can life with it...

File: user-info-provider/src/main/java/de/intension/mapper/oidc/UserInfoProviderMapper.java (hunk 5 above)
Comment (mkunz-intension) [reply user-info-provider/src/main/java/de/intension/mapper/oidc/UserInfoProviderMapper.java:111]: Sure

File: user-info-api/README.md (hunk 1 above)
Comment (mkunz-intension) [reply user-info-api/README.md:1]: done

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,10 @@
 # Change Log
 All notable changes to this project will be documented here.
 
+## Version [3.1.0]
+
+- New extensions to request and support SANIS specific fields
```
[the remark this answers is not in the document]
Comment (mkunz-intension) [reply CHANGELOG.md:6]: agree

File: user-info-api/README.md (hunk 3 above)
Comment (ingokuba) [reply user-info-api/README.md:43]: Aha :smile: no time to understand :runner: 
https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form

File: user-info-api/src/main/java/de/intension/api/UserInfoAttribute.java (hunk 4 above)
Comment (ingokuba) [reply user-info-api/src/main/java/de/intension/api/UserInfoAttribute.java]: True - noone likes the keycloak.v2 admin UI :grin:

File: user-info-provider/src/main/java/de/intension/mapper/oidc/UserInfoProviderMapper.java (hunk 5 above)
Comment (ingokuba) [reply user-info-provider/src/main/java/de/intension/mapper/oidc/UserInfoProviderMapper.java:111]: No localization - but fine for now hm :+1:

**Review by ingokuba — APPROVED** [review]

**Pull request merged**